src/test/java/sender/TestSender.java

```diff
@@ -0,0 +1,40 @@
+package sender;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.util.ArrayList;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestSender {

```

Comment:
These tests will pass even if the output format is wrong. Focus on testing the sender functionality - the most crucial is the output format.
In the absence of a test for the output format, any mistakes will only be caught only in integration.